sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -530,12 +530,27 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] extends Product {
     s"$nodeName ${argString(maxFields)}".trim
   }
 
+  def simpleString(planToOperatorID: mutable.LinkedHashMap[TreeNode[_], Int]): String = {
+    val operatorID = planToOperatorID.get(this).map(v => s"$v").getOrElse("unknown")
+    s"$nodeName ($operatorID)".trim
+  }
+
   /** ONE line description of this node with more information */
   def verboseString(maxFields: Int): String
 
   /** ONE line description of this node with some suffix information */
   def verboseStringWithSuffix(maxFields: Int): String = verboseString(maxFields)
 
+  def verboseString(
+      planToOperatorID: mutable.LinkedHashMap[TreeNode[_], Int],
```

Comment:
if this is specific to query plan, we can put it in `QueryPlan` instead of `TreeNode`.

Reply:
@cloud-fan Sounds reasonable. Its been a while :-) Let me try to look at the code later today.